src/dsi.rs

```diff
@@ -0,0 +1,693 @@
+//! Display Serial Interface
+//!
+//! Interface with MIPI D-PHY
+
+use crate::ltdc::DisplayConfig;
+use crate::rcc::{Clocks, Enable};
+use crate::{pac::DSI, time::Hertz};
+use core::cmp::{max, min};
+use embedded_display_controller::dsi::{DsiHostCtrlIo, DsiReadCommand, DsiWriteCommand};
+//use embedded_display_controller::DisplayConfiguration;
+#[cfg(feature = "log")]
```

Comment:
There is no `log` feature in f4xx-hal. Unless you want to add it.

Reply:
True; this came from the h7xx-hal so I can look at adding it in here, or will remove the code using it if that gets out of scope.

Reply:
I removed the logging code, along with some other leftover debug things I inadvertently left in.

It would be interesting to have an `stm-dsi` crate at some point, given the DSI block is the same across the f4/f7/h7 families (or close enough to `cfg`ify), and with that some of the other elements could be added.

Reply:
For note: I'm experimenting with merge f4 and f7 hals as most close to each other: https://github.com/stm32-rs/stm32f4xx-hal/pull/667 . But I need testers as I do not have F7 devices.

Reply:
That is interesting, and quite an endeavor... in such cases would a new crate be warranted, if only to have a name which reflects the combined support?  I think we'd want any future F7 and F4 HAL work to go in only one place.

I don't have any F7 devices currently but will be happy to test things if I wind up getting one.